apply boot-2.7.0 and add logback-classic-db

btonokit opened the pull request:
Please review #981,#982

relate to [TSF4J5-1301](https://terasolunaorg.atlassian.net/browse/TSF4J5-1301)([TSF4J5-1289](https://terasolunaorg.atlassian.net/browse/TSF4J5-1289)),[TSF4J5-1365](https://terasolunaorg.atlassian.net/browse/TSF4J5-1365),

btkawaharahd commented on terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/FunctionTestSupport.java:
```diff
@@ -16,6 +16,7 @@
+import java.time.Duration;
 import java.util.HashSet;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
```
Unnecessary import statements remain.

btonokit replied:
40bc931b4a8e961bbca35c58395e7176a7340595
deleted the unnecessary import statement .

btonokit commented:
60de9a21d6c31f4197d1a7864ef71bbb92b593f9
Fixed missing deprecated API.
bcf83692d6128e3d166de2ca9991483649e79b4a
Removed unnecessary listener classes that used deprecated API.
Since gfw-functionaltest does not test IE and support for IE has already ended, I deleted the InternetExplorerDriver branch statement.

yoshikawaa commented on terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/WebDriverOperations.java:
```diff
@@ -55,7 +55,7 @@ public void setDefaultTimeoutForImplicitlyWait(
     public boolean exists(By by) {
         webDriver.findElement(By.tagName("body"));
-        setTimeoutForImplicitlyWait(0, TimeUnit.SECONDS);
+        setTimeoutForImplicitlyWait(Duration.ofSeconds(0));
```
`Duration.Zero` is not fine?

btkatoufm replied:
Use the Duration.Zero provided.

btonokit replied:
Fixed a20f7a7c18f13432b4cbb129a57a352704932c3b in https://github.com/terasolunaorg/terasoluna-gfw-functionaltest/pull/986

yoshikawaa commented on terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/transactiontoken/TransactionTokenTest.java:
```diff
@@ -69,8 +59,6 @@ public void setUp() {
         }
         bootDefaultWebDriver();
-        driver = new EventFiringWebDriver(getDefaultWebDriver());
-        driver.register(new ScreenCaptureWebDriverEventListener(screenCapture));
```
`EventFiringWebDriver` & `WebDriverEventListener` has bean deprecated, but should you remove the screen capture feature?

btkatoufm replied:
No captures were taken because `selenium.enableCapture=false`. There is no visual confirmation of the capture in the test item either. Also, since the captcha is obtained in a separate class when the test fails, we believe it is safe to remove the `ScreenCaptureWebDriverEventListener`.

yoshikawaa replied:
`ScreenCaptureWebDriverEventListener` captures after click and after page transition, not on test failure. There is no perfect replacement by another mechanism.
Whether or not this mechanism is necessary should be discussed in a new Issue. At least I don't think the presented reason are good enough to remove the mechanism itself.（I'm not strongly against removing it.）

btonokit replied:
Created #985

yoshikawaa commented on terasoluna-gfw-functionaltest-web/pom.xml:
```diff
@@ -154,6 +154,13 @@
+      <groupId>ch.qos.logback.db</groupId>
+      <artifactId>logback-classic-db</artifactId>
+    </dependency>
+    <!-- == End logback-db == -->
```
Why do you need to re-define the domain dependencies?

btkatoufm replied:
Oh...No redefinition necessary.

btonokit replied:
Fixed 16ad9f9d0a10b10f96c8e6a31e426eb18d66af96 in #986.

yoshikawaa commented on terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/FunctionTestSupport.java:
```diff
@@ -187,14 +187,15 @@ protected void bootDefaultWebDriver() {
                 defaultTimeoutSecForImplicitlyWait);
-        this.webDriverWait = new WebDriverWait(driver, defaultTimeoutSecForImplicitlyWait);
+        this.webDriverWait = new WebDriverWait(driver, Duration.ofSeconds(
+                defaultTimeoutSecForImplicitlyWait));
```
`Duration.ofSeconds(defaultTimeoutSecForImplicitlyWait)` is used multiple times, so it is better to define a variable.

btkatoufm replied:
`Duration.ofSeconds` internal implementation would create instances needlessly, so let's cut it out into a variable.

btonokit replied:
Fixed with a20f7a7c18f13432b4cbb129a57a352704932c3b in #986

yoshikawaa reviewed (commented):
Trivial comments, regards.

btkatoufm commented on pom.xml:
```diff
@@ -241,6 +249,7 @@
     <ojdbc.version>21.3.0.0</ojdbc.version>
     <db2jcc4.version>4.21.29</db2jcc4.version>
     <webdrivermanager.version>3.1.1</webdrivermanager.version>
+    <logback-classic-db.version>1.2.11.1</logback-classic-db.version>
```
We believe that logback-classic-db should be an exceptional property, since the numbering system is different here than in other places due to logback's various circumstances.

yoshikawaa replied:
For example, logback may be continuously released, but logback-db may not be, so I agree your opinion.
However, the property name prefix would be better for the project or group `logback-db` rather than the artifact `logback-classic-db`. Thanks.

btkatoufm replied:
Yes, let's say logback-db.

btonokit replied:
Fixed with 01cb6fd6091321d21e9ecd2e52918755fc0d26e6 in https://github.com/terasolunaorg/terasoluna-gfw-functionaltest/pull/986
